contracts/contracts/GrantRegistry.sol

```diff
@@ -1,135 +1,132 @@
 pragma solidity ^0.8.5;
 
+/**
+ * @notice The Gitcoin GrantRegistry contract keeps track of all grants that have been created.
+ * It is designed to be a singleton, i.e. there is only one instance of a GrantRegistry which
+ * tracks all grants. It behaves as follows:
+ *   - Anyone can create a grant by calling `createGrant`
+ *   - A grant's `owner` can edit their grant using the `updateGrant` family of method
+ *   - The `getAllGrants` and `getGrants` view methods are used to fetch grant data
+ */
 contract GrantRegistry {
-  // Grant ID - The hash of the grants meetadata
-  bytes32 public grantRegistryId;
-  // Metadata Pointer
-  string public metaPtr;
-
-  // Owner
-  address owner;
-
-  modifier onlyOwner() {
-    require(msg.sender == owner);
-    _;
-  }
-
-  // TODO: Only allow the owner (which is the curator) address to configure the contract
-  // while its in config mode, then switch to active. Once active, the owner can only
-  // close the registry (have a conversation around this approach)
-  constructor(
-    address _owner,
-    bytes32 _grantRegistryId,
-    string memory _metaPtr
-  ) {
-    owner = _owner;
-    //state = State.CONFIG;
-    grantRegistryId = _grantRegistryId;
-    metaPtr = _metaPtr;
-  }
-
-  // State
-  enum State {
-    CONFIG,
-    ACTIVE,
-    CLOSED
-  }
-  //State public state;
-
-  // Grant Pointer Object
-  // Mapping from GrantID to set of Owner Address, Payout Address, Metadata Pointer, Replaces Grant (optional)
-  // TODO: Implement safemath for avoiding overflows with ints
-  uint256 public grantCount = 0;
-  mapping(bytes32 => Grant) public grants;
-  bytes32[] public grantIds;
+  // --- Data ---
+  /// @notice Number of grants stored in this registry
+  uint96 public grantCount;
```

Comment:
why do we store this value btw? 
Can this not be something which derived via a subgraph or so ?

Reply:
The `grantCount` is used to determine the grants ID in the `createGrant` method, which is why we store it